From 21c5b3db52085807871b77f70ed54d806a741681 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Fri, 13 Sep 2024 11:26:54 +0200
Subject: [PATCH 1224/1695] dts: stm32: replace resets C2_R and C2_HOLDBOOT_R

Replaces theses resets by the same ones controlled by m33_rproc driver.
This driver implement a specific filtering.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 21370b8c439768245a8b2b778bda26cf6db01004 dts: stm32: replace resets C2_R and C2_HOLDBOOT_R]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: Ic427629992622dc61667dc5b7bf15db1f58b9127
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404700
Tested-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 8 ++++----
 core/arch/arm/dts/stm32mp231.dtsi | 8 ++++----
 core/arch/arm/dts/stm32mp251.dtsi | 8 ++++----
 3 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index cd7aae372..8b872e297 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -965,7 +965,7 @@
 					reset@1 {
 						reg = <RST_SCMI_C2_R>;
 						domain-name = "c2";
-						resets = <&rcc C2_R>;
+						resets = <&m33_rproc 0>;
 					};
 
 					reset@2 {
@@ -977,7 +977,7 @@
 					reset@3 {
 						reg = <RST_SCMI_C2_HOLDBOOT_R>;
 						domain-name = "c2_holdboot";
-						resets = <&rcc C2_HOLDBOOT_R>;
+						resets = <&m33_rproc 1>;
 					};
 
 					reset@4 {
@@ -1858,7 +1858,7 @@
 					reset@1 {
 						reg = <RST_SCMI_C2_R>;
 						domain-name = "c2";
-						resets = <&rcc C2_R>;
+						resets = <&m33_rproc 0>;
 					};
 
 					reset@2 {
@@ -1870,7 +1870,7 @@
 					reset@3 {
 						reg = <RST_SCMI_C2_HOLDBOOT_R>;
 						domain-name = "c2_holdboot";
-						resets = <&rcc C2_HOLDBOOT_R>;
+						resets = <&m33_rproc 1>;
 					};
 
 					reset@4 {
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 9572b592b..746e88dd2 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -1065,7 +1065,7 @@
 					reset@1 {
 						reg = <RST_SCMI_C2_R>;
 						domain-name = "c2";
-						resets = <&rcc C2_R>;
+						resets = <&m33_rproc 0>;
 					};
 
 					reset@2 {
@@ -1077,7 +1077,7 @@
 					reset@3 {
 						reg = <RST_SCMI_C2_HOLDBOOT_R>;
 						domain-name = "c2_holdboot";
-						resets = <&rcc C2_HOLDBOOT_R>;
+						resets = <&m33_rproc 1>;
 					};
 
 					reset@4 {
@@ -2062,7 +2062,7 @@
 					reset@1 {
 						reg = <RST_SCMI_C2_R>;
 						domain-name = "c2";
-						resets = <&rcc C2_R>;
+						resets = <&m33_rproc 0>;
 					};
 
 					reset@2 {
@@ -2074,7 +2074,7 @@
 					reset@3 {
 						reg = <RST_SCMI_C2_HOLDBOOT_R>;
 						domain-name = "c2_holdboot";
-						resets = <&rcc C2_HOLDBOOT_R>;
+						resets = <&m33_rproc 1>;
 					};
 
 					reset@4 {
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 7597d44a4..8c0eaef57 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1065,7 +1065,7 @@
 					reset@1 {
 						reg = <RST_SCMI_C2_R>;
 						domain-name = "c2";
-						resets = <&rcc C2_R>;
+						resets = <&m33_rproc 0>;
 					};
 
 					reset@2 {
@@ -1077,7 +1077,7 @@
 					reset@3 {
 						reg = <RST_SCMI_C2_HOLDBOOT_R>;
 						domain-name = "c2_holdboot";
-						resets = <&rcc C2_HOLDBOOT_R>;
+						resets = <&m33_rproc 1>;
 					};
 
 					reset@4 {
@@ -2057,7 +2057,7 @@
 					reset@1 {
 						reg = <RST_SCMI_C2_R>;
 						domain-name = "c2";
-						resets = <&rcc C2_R>;
+						resets = <&m33_rproc 0>;
 					};
 
 					reset@2 {
@@ -2069,7 +2069,7 @@
 					reset@3 {
 						reg = <RST_SCMI_C2_HOLDBOOT_R>;
 						domain-name = "c2_holdboot";
-						resets = <&rcc C2_HOLDBOOT_R>;
+						resets = <&m33_rproc 1>;
 					};
 
 					reset@4 {
-- 
2.47.2

